spec/controllers/churn_controller_spec.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe ChurnController, type: :controller do
+  let(:user) { create(:user) }
+  let(:product) { create(:subscription_product, user: user) }
+
+  before do
+    sign_in user
+  end
+
+  describe "GET #show" do
+    context "when user has subscription products" do
+      before { product }
```

Comment:
You can just eager load this with `let!` instead of invoking it in a before block